domain/machine/service/service.go

```diff
@@ -7,12 +7,14 @@ import (
 	"context"
 
 	"github.com/juju/errors"
+
+	"github.com/juju/juju/internal/uuid"
 )
 
 // State describes retrieval and persistence methods for machines.
 type State interface {
 	// UpsertMachine persists the input machine entity.
-	UpsertMachine(context.Context, string) error
+	UpsertMachine(context.Context, string, string, string) error
```

Comment:
Can you rename this to `CreateMachine`. We know that we're creating a machine, no need to pretend that we might be creating vs updating it.